### ISSUE TITLE
[feature request]: Get contributors data from Github API 

### ISSUE THREAD

**Issue by ashutoshkrris** [ashutoshkrris/EazyLoader#11]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Describe the feature.

As we are building a contributors page, we need data of contributors.

### Problem/Motivation.

As people are gradually contributing to the project, we must be thankful for their efforts.

### Possible Solution/Pitch.

There are two approaches : 

1. Use a JSON file such as `contributors.json` and then using Javascript, we can make use of that data.
2. Use Github API to get the list of contributors and their details to directly serve it from backend.

### Anything else?

Since some bots such as `dependabots` can be found in the contributors' list. So, while using Github API, make sure you exclude them.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Tawishi** [comment]
hey @ashutoshkrris. I would like to work on this issue

**Comment by ashutoshkrris** [comment]
Sure @Tawishi , go ahead!

**Comment by Tawishi** [comment]
I am able to fetch the contributors, excluding the bot, using the Github API
where do we want these to be added in the backend?

**Comment by ashutoshkrris** [comment]
you can create a file called `contributors.py` within the core package @Tawishi 
also, if possible use Github access tokens to increase the number of requests

**Comment by Tawishi** [comment]
ok, so getting the `avatar_url` and `html_url` is done
looking for a way to get the contributor's name

**Comment by ashutoshkrris** [comment]
This API might be useful : `https://api.github.com/users/{username}`

**Comment by Tawishi** [comment]
this works @ashutoshkrris 
we have the  `avatar_url` `html_url` and `name` now
so contributors.py has 2 functions - 
- for getting contributors `avatar_url`, `html_url`, `github username`
- for getting name of contributor using `github username`

**Comment by Tawishi** [comment]
almost forgot that. working on it

**Comment by Tawishi** [comment]
@ashutoshkrris why use tokens for these allow-any get requests ?
>@Tawishi also, if possible use Github access tokens to increase the number of requests

how do tokens increase the number of requests?

**Comment by ashutoshkrris** [comment]
Hey @Tawishi 
Head over to https://github.com/ashutoshkrris/EazyLoader/discussions/13 and let's discuss this

**Comment by Tawishi** [comment]
i got the answer in the docs that is why I removed the comment @ashutoshkrris 
Thank you for the update, anyways

**Comment by Tawishi** [comment]
we will use whose personal token here?

**Comment by ashutoshkrris** [comment]
You can create one for yourself and try with that.
Make sure, you don't commit that. You can put that in the `.env` file
For the project deployment, I'll use mine

**Comment by Tawishi** [comment]
what should the data be returned as, any specific return type of the required contributor information like JSON, etc.?

**Comment by ashutoshkrris** [comment]
you can just return a python dictionary from your function or method.

**Comment by Tawishi** [comment]
ok

**Comment by ashutoshkrris** [comment]
you can return like this : 
```python
result = {
    "username": username,
    "name": name,
    "profile_url": html_url
}
return result
```

**Comment by Tawishi** [comment]
all done. I will create a PR
Please ping here in case changes are needed.

**Comment by Tawishi** [comment]
does this need any documentation update?

**Comment by ashutoshkrris** [comment]
Nope, I don't think so as the user need not perform any extra step for getting added as contributors

**Comment by Tawishi** [comment]
Thank you @ashutoshkrris for your sypport and patience

**Comment by ashutoshkrris** [comment]
Thanks for contributing @ashutoshkrris 
Feel free to add more features, and create issues
